pkg/devcontainer/setup/setup.go

```diff
@@ -336,17 +339,65 @@ func runPostCreateCommand(commands []types.LifecycleHook, user, dir string, remo
 			cmd.Dir = dir
 			cmd.Env = os.Environ()
 			cmd.Env = append(cmd.Env, remoteEnvArr...)
-			cmd.Stdout = writer
-			cmd.Stderr = errwriter
-			log.Debugf("Executing command %s: %s...", k, cmd.Args)
-			err := cmd.Run()
+
+			// Create pipes for stdout and stderr
+			stdoutPipe, err := cmd.StdoutPipe()
+			if err != nil {
+				return fmt.Errorf("failed to get stdout pipe: %w", err)
+			}
+			stderrPipe, err := cmd.StderrPipe()
+			if err != nil {
+				return fmt.Errorf("failed to get stderr pipe: %w", err)
+			}
+
+			// Start the command
+			if err := cmd.Start(); err != nil {
+				return fmt.Errorf("failed to start command: %w", err)
+			}
+
+			// Use WaitGroup to wait for both stdout and stderr processing
+			var wg sync.WaitGroup
+			wg.Add(2)
+
+			go logPipeOutput(log, stdoutPipe, logrus.InfoLevel, &wg)
+			go logPipeOutput(log, stderrPipe, logrus.ErrorLevel, &wg)
+
+			// Wait for command to finish
+			wg.Wait()
```

Comment:
there is no real interaction, command starts separately before we start routines that process piped output from this command. And we need to ensure that all reads from pipe finished before closing the pipe: from cmd.StdoutPipe:
```
Wait will close the pipe after seeing the command exit, so most callers need not close the pipe themselves. It is thus incorrect to call Wait before all reads from the pipe have completed. 
```
in case of some buffer failure scanner should pick it up, exit with error and do wg.Done

Reply:
Ah nice, thanks!